numba/types/__init__.py

```diff
@@ -73,6 +73,8 @@
 complex_domain = frozenset([complex64, complex128])
 number_domain = real_domain | integer_domain | complex_domain
 
+index_type = integer_domain
```

Comment:
Not sure if this should go here or in `numba/listobject.py`. If it does stay here, it should probably be equipped with a comment.